### PR TITLE
Feature: Added Onboarding Page

### LINKED ISSUE #23: Include a welcome (onboarding) page for users
**Describe the feature you'd like**

Include a welcome (onboarding) page for users

### LINKED ISSUE #57: [BUG] Scrolling issue in Media Details Page.
**Summary:**

Profile Page scrolling is not smooth.

**Steps to reproduce:**

- Login to the app.
- Open any courses and go to uploads section.
- Try scrolling in the category section of the page. 

**Device and Android version:**

**Model No:** Moto G5S+
**Android version:** 8.1.1

**Would you like to work on the issue?**
No

### PR DESCRIPTION
**Pull request by gconnect**
**What does this PR do?**
Fixes #23 #83 #98 #97 
Fixes #57 #76 

- User can now have access to the onboarding Screens
- User can now select either of WikiEdu and Outreach dashboard with ease when during login or signup

**What Should be done for the PR to work?**
- Already logged in logout/ clear cache so you can view the onboarding

### CODE REVIEW

**Comment by gconnect** [comment]
@ujjwalagrawal17 please review. Thanks

**Comment by ujjwalagrawal17** [comment]
Can you also add the before, after screenshots. We need to follow https://github.com/WikiEducationFoundation/apps-android-wikiedudashboard/blob/master/PULL_REQUEST_TEMPLATE.md from now on.

**Comment by gconnect** [comment]
I have completed the onboarding and sent pull request. Here are the
screenshots of before and after.

**Comment by gconnect** [comment]
I have worked on the comments and pushed them. Please review. Thanks

**Comment by gconnect** [comment]
I have updated them with the code checks comments. Please review

**Comment by ujjwalagrawal17** [comment]
Can you also replace !! with ? as we want to make sure we don't write code that may cause #90 in future.

**Comment by ujjwalagrawal17** [comment]
Also, Can you please look into the codacy PR Quality review and implement the changes suggested by it.

**Comment by ujjwalagrawal17** [comment]
Please remove old welcome activity if it's not needed anymore.

**Comment by ujjwalagrawal17** [comment]
Can you explain why do we need welcome activity ? 
I guess the flow of the app now is.
1. Splash Activity
2. WelcomeHostActivity (4 fragments)
3. HomeActivity (redirected after login)

**Comment by gconnect** [comment]
All code checks have passed and conflict resolved

**Comment by ujjwalagrawal17** [comment]
I have added few comments in the code. Can you implement them.

**Comment by ujjwalagrawal17** [comment]
There are changes in may files (43). We should not change even the indentation of the files if they are not related to the issue. Otherwise, it becomes difficult for the reviewer to review.

**Comment by gconnect** [comment]
I have made the necessary modifications

**Comment by gconnect** [comment]
Please review.Thanks.

**Comment by gconnect** [comment]
The welcome Activity is because after the onboarding pages once you click get started it launches the welcome activity where the user can choose between wiki or outreach dashboard to signup or login. I wanted to change the welcome Activity to Welcome fragment but realized welcome Activity was also referenced from the settings activity and I didn't want to tamper with the settings Activity. So I had to revert back to the welcome Activity. Either ways the get started button launch the activity

**Comment by ujjwalagrawal17** [comment]
Just tested the PR, the new Welcome UI, both the login screens looks great! few minor issues in code need to be resolved then we can merge this PR.

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/welcome/onboarding/page/OnboardingPage.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * or sale is expressly withheld.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
```
[the remark this answers is not in the document]
Comment (ujjwalagrawal17) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/welcome/onboarding/page/OnboardingPage.kt]: @gconnect can you remove the above comment? It is not relevant to our project.

**Comment by ujjwalagrawal17** [comment]
@gconnect can you resolve the merge conflicts, remove the license. Looking forward to merging this.

**Comment by gconnect** [comment]
Ready to be merged. Conflict fixed and license removed.

**Comment by gconnect** [comment]
@kulloveth Thanks. I didn't notice it on time. It has been fixed. Everything is okay now.
@ujjwalagrawal17, please review. Thanks.

**Comment by ujjwalagrawal17** [comment]
Nice Job !

**Comment by gconnect** [comment]
Thanks @ujjwalagrawal17  for the review.

**Pull request merged**